tmt/steps/discover/fmf.py

```diff
@@ -407,8 +407,12 @@ def assert_git_url(plan_name: Optional[str] = None) -> None:
                     dist_git_extract = sourcedir
                 else:
                     try:
-                        dist_git_extract = glob.glob(os.path.join(
-                            sourcedir, dist_git_extract.lstrip('/')))[0]
+                        # Glob everything
+                        glob_path = glob.glob(os.path.join(sourcedir, dist_git_extract.lstrip('/')))
+                        # Use only the directories
+                        glob_dirs = [d for d  in glob_path if Path(d).is_dir()]
+                        # Use only the first valid directory
+                        dist_git_extract = glob_dirs[0]
```

Comment:
`os.path.join` is a left-over from our recent transition to `pathlib.Path`, let's get rid of it :)

How about this?

```python
glob_dirs = [
    child
    for child in sourcedir.glob(dist_git_extract.lstrip('/'))
    # Use only the directories
    if child.is_dir()
]

# Use only the first valid directory
dist_git_extract = glob_dirs[0]
```

Reply:
I've had a conference and a cold so I didn't work on this, but I think we should change the logic here:
- no `path` is passed use the search algorithm
- if `path` is passed it has to be relative to `BUILDROOT`
  - in this regard dist_git_extract should not take only the 1st glob_dirs, but copy all files and glob patterns
  - allow for globbing for the path, i.e. one should pass `scikit-build-core-*/path/to/fmf_root`

Reply:
TBH, the whole function is hard to navigate for me, therefore I can't comment too much on the change in its workflow. I was only after `os.path.join` in this case :) Let's ping @lukaszachy who `git blame`s for these lines.